modules/caddyhttp/autohttps.go

```diff
@@ -480,6 +486,22 @@ func (app *App) createAutomationPolicies(ctx caddy.Context, internalNames []stri
 		app.tlsApp.Automation = new(caddytls.AutomationConfig)
 	}
 	for _, ap := range app.tlsApp.Automation.Policies {
+		// on-demand policies can have the tailscale manager added implicitly
+		// if there's no explicit manager configured -- for convenience
+		if ap.OnDemand && len(ap.Managers) == 0 {
+			var ts caddytls.Tailscale
+			if err := ts.Provision(ctx); err != nil {
+				return err
```

Comment:
Shouldn't this check for `len(tailscaleNames)` or something? I'm not sure tailscale should be provisioned unless there's matching names

Reply:
On-demand TLS doesn't know the names beforehand.

The Tailscale manager can only get certs from Tailscale's `.ts.net` domain, so it's a no-op even if we add it in all cases.